sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -115,7 +120,40 @@ public HttpLoggingPolicy(HttpLogDetailLevel detailLevel, boolean prettyPrintJSON
         return reqBodyLoggingMono;
     }
 
-    private Function<HttpResponse, Mono<HttpResponse>> logResponseDelegate(final ClientLogger logger, final URL url,
+    private void formatAllowableHeaders(Set<String> allowedHeaderNames, HttpRequest request, ClientLogger logger) {
+        for (HttpHeader header : request.getHeaders()) {
+            if (allowedHeaderNames.contains(header)) {
+                logger.info(header.toString());
+            } else {
+                logger.info(REDACTED_PLACEHOLDER);
+            }
+        }
+    }
+
+    private void formatAllowableQueryParams(Set<String> allowedQueryParamNames, HttpRequest request,
+                                            ClientLogger logger) {
+        StringBuilder sb = new StringBuilder();
+        String queryString = request.getUrl().getQuery();
+        String[] queryParams = queryString.split("&");
+        for (String queryParam : queryParams) {
+            String[] queryPair = queryParam.split("=");
+            if (2 != queryPair.length) {
+                throw new IllegalArgumentException("Parse failed for " + queryString);
+            }
```

Comment:
It would be better if not do any request validation (and throwing of errors) in logging framework. I would rather do:

```java
 String[] queryPair = queryParam.split("=", 2);
 if (queryPair.length == 2) {
   if(allowedQueryParamNames.contains(queryPair[0])) {
       sb.append(queryParam);
   } else {
       sb.append(queryPair[0]).append("=").append(REDACTED_PLACEHOLDER);
   }
 } else {
   sb.append(queryParam);
 }
```

this way we leave parameters like `c` and `d` in `?a=b&c=&d` as it is.

Reply:
Yeah, even I was thinking of the same to not do any validation. But just put out there to know it better. Will remove it, 👍